### PR TITLE
😱  🚀  🎨  tests: use truncate instead of database deletion

### PR DESCRIPTION
**Pull request by kirrg001**
refs #7718, refs #7470

- should bring massive speed improvement
- **could** also fix random test failures (e.g. sqlite database is busy)
- lot's of auto formatting, **sorry for that (!)**, but i think we should no longer have huge spaces between variables and different formats - but can also revert!

The core change in this PR is the modification of the teardown test helper: https://github.com/TryGhost/Ghost/pull/8119/files#diff-085888ce4e458820a35b19844ba3d990R721

I was very annoyed of the slowness of mysql tests. So i just spend 2 hours hopefully fixing this.
We'll see that Travis is saying...

#### BEFORE
before

#### AFTER
after_no_cache

after1

😜 💦 🚑

It reduces the MySQL test speed massively. 
**The tests are running locally way, way faster**, this is because Travis need's time to create the container etc. etc. Not sure it reduces random failures. I ran it 3 times, not random failure....but we'll see.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -518,7 +531,7 @@ var overrides      = require('./core/server/overrides'),
         // Unit tests do **not** touch the database.
         // A coverage report can be generated for these tests using the `grunt test-coverage` task.
         grunt.registerTask('test-unit', 'Run unit tests (mocha)',
-            ['test-setup', 'mochacli:unit']
+            ['knex-migrator', 'test-setup', 'mochacli:unit']
```
Comment (ErisDS) [inline Gruntfile.js]: Do we need this here for unit tests?! Unit tests should not touch the DB 😉

File: Gruntfile.js [hunk 2]
```diff
@@ -546,7 +559,7 @@ var overrides      = require('./core/server/overrides'),
         //
         // A coverage report can be generated for these tests using the `grunt test-coverage` task.
         grunt.registerTask('test-integration', 'Run integration tests (mocha + db access)',
-            ['test-setup', 'mochacli:integration']
+            ['knex-migrator', 'test-setup', 'mochacli:integration']
```
Comment (ErisDS) [inline Gruntfile.js]: Would it be possible to move this into the test-setup command?

**Review by ErisDS — CHANGES_REQUESTED** [review]
This is awesome 🎉 !!!!

I have a teeny-tiny comment about the changes to the Gruntfile.js - I do think it would make more sense to add knex-migrator as part of the test-setup task? 

Otherwise let's merge this asapppppppppppp!

File: Gruntfile.js (hunk 2 above)
Comment (kirrg001) [inline Gruntfile.js]: yes agree! did that very quick to see that it's working. good suggestion

File: Gruntfile.js (hunk 1 above)
Comment (kirrg001) [inline Gruntfile.js]: i think one test is using the db and i have to admit, i have written this test 😭😂
will double-check

File: Gruntfile.js (hunk 2 above)
Comment (ErisDS) [inline Gruntfile.js]: 👍

File: Gruntfile.js (hunk 1 above)
Comment (ErisDS) [inline Gruntfile.js]: Ok, if we move knex-migrator into test-setup it will get run here anyway, can fix later.

Comment (kirrg001) [inline Gruntfile.js]: true 👍

**Comment by kirrg001** [comment]
Updated

**Review by ErisDS — APPROVED** [review]

**Comment by ErisDS** [comment]
AWESOME 🚀

**Pull request merged**